============================= test session starts ==============================
platform linux -- Python 3.10.19, pytest-8.4.2, pluggy-1.6.0 -- /opt/hostedtoolcache/Python/3.10.19/x64/bin/python
cachedir: .pytest_cache
rootdir: /home/runner/work/iris_dvc_pipeline_mlops/iris_dvc_pipeline_mlops
plugins: hydra-core-1.3.2
collecting ... collected 5 items

tests/test_validation_and_evaluation.py::test_data_file_exists PASSED    [ 20%]
tests/test_validation_and_evaluation.py::test_data_columns_and_nulls PASSED [ 40%]
tests/test_validation_and_evaluation.py::test_model_file_exists PASSED   [ 60%]
tests/test_validation_and_evaluation.py::test_model_prediction_shape PASSED [ 80%]
tests/test_validation_and_evaluation.py::test_metrics_file_and_accuracy PASSED [100%]

=============================== warnings summary ===============================
tests/test_validation_and_evaluation.py::test_model_prediction_shape
  /opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/sklearn/base.py:493: UserWarning: X does not have valid feature names, but DecisionTreeClassifier was fitted with feature names
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================= 5 passed, 1 warning in 1.11s =========================
